src/app/teacher/teacher-run-list/teacher-run-list.component.ts

```diff
@@ -41,19 +44,35 @@ export class TeacherRunListComponent implements OnInit {
   }
 
   private getRuns(): void {
-    this.teacherService.getRuns().subscribe((runs) => {
-      const userId = this.userService.getUserId();
-      this.runs = runs.map((run) => {
-        const teacherRun = new TeacherRun(run);
-        teacherRun.shared = !teacherRun.isOwner(userId);
-        return teacherRun;
+    this.teacherService
+      .getRuns(this.FETCH_RECENT_RUNS_MAX)
+      .pipe(mergeMap((runs) => this.processRecentRuns(runs)))
+      .subscribe((runs: TeacherRun[]) => {
+        this.setRuns(runs);
+        this.processRuns();
+        this.loaded = true;
       });
-      this.processRuns();
-      this.loaded = true;
+  }
+
+  private processRecentRuns(runs: TeacherRun[]): Observable<TeacherRun[]> {
+    if (runs.length < this.FETCH_RECENT_RUNS_MAX) {
+      return of(runs);
+    } else {
+      this.setRuns(runs);
+      return this.teacherService.getRuns();
+    }
+  }
+
+  private setRuns(runs: TeacherRun[]) {
+    this.runs = runs.map((run) => {
+      const teacherRun = new TeacherRun(run);
+      teacherRun.shared = !teacherRun.isOwner(this.userService.getUserId());
+      return teacherRun;
     });
+    this.filteredRuns = this.runs;
   }
 
-  private subscribeToRuns(): void {
+  private subscribeToRuns(): any {
```

Comment:
Should the subscribeToRuns() return type be void since it isn't returning anything?